src/erlcloud_ddb_impl.erl

```diff
@@ -68,23 +68,24 @@
         ]).
 
 %% Internal impl api
--export([request/3]).
+-export([request/4]).
 
 -export_type([json_return/0, attempt/0, retry_fun/0]).
 
--type json_return() :: ok | {ok, jsx:json_term()} | {error, term()}.
+-type json_return() :: ok | {ok, jsx:json_term()} | {error, term()} | list().
```

Comment:
`list()` seems rather vague, right? Why not a tuple with an indication of what the actual return value means?

Reply:
why wouldn't this return `{ok, {RequestHeaders, Body}}` anyway?

Reply:
maybe `{ok, {request, Headers, Body}}`, for even more clarity? This seems like an OK approach, but the output format is extremely ambiguous and out of step with the style here.

Reply:
Yes, it makes it much more clear and goes within erlcloud style. I changed that for the sake of clarity 😄